src/annotated/marked_yaml.rs

```diff
@@ -0,0 +1,152 @@
+    ///
+    /// See the function [`load_from_str`] for more details.
+    ///
+    /// # Errors
+    /// Returns `ScanError` when loading fails.
+    ///
+    /// [`load_from_str`]: `Yaml::load_from_str`
+    pub fn load_from_str(source: &str) -> Result<Vec<Self>, ScanError> {
+        Self::load_from_iter(source.chars())
+    }
+
+    /// Load the contents of the given iterator as an array of YAML documents.
+    ///
+    /// See the function [`load_from_str`] for more details.
+    ///
+    /// # Errors
+    /// Returns `ScanError` when loading fails.
+    ///
+    /// [`load_from_str`]: `Yaml::load_from_str`
+    pub fn load_from_iter<I: Iterator<Item = char>>(source: I) -> Result<Vec<Self>, ScanError> {
+        let mut parser = Parser::new(source);
+        Self::load_from_parser(&mut parser)
+    }
+
+    /// Load the contents from the specified [`Parser`] as an array of YAML documents.
+    ///
+    /// See the function [`load_from_str`] for more details.
+    ///
+    /// # Errors
+    /// Returns `ScanError` when loading fails.
+    ///
+    /// [`load_from_str`]: `Yaml::load_from_str`
+    pub fn load_from_parser<I: Iterator<Item = char>>(
+        parser: &mut Parser<I>,
+    ) -> Result<Vec<Self>, ScanError> {
+        let mut loader = YamlLoader::<Self>::default();
+        parser.load(&mut loader, true)?;
+        Ok(loader.into_documents())
+    }
+}
+
+impl PartialEq for MarkedYaml {
+    fn eq(&self, other: &Self) -> bool {
+        self.data.eq(&other.data)
+    }
+}
+
+// I don't know if it's okay to implement that, but we need it for the hashmap.
+impl Eq for MarkedYaml {}
+
+impl std::hash::Hash for MarkedYaml {
+    fn hash<H: std::hash::Hasher>(&self, state: &mut H) {
+        self.data.hash(state);
+    }
+}
+
+impl From<YamlData<MarkedYaml>> for MarkedYaml {
+    fn from(value: YamlData<MarkedYaml>) -> Self {
+        Self {
+            marker: Marker::default(),
+            data: value,
+        }
+    }
+}
+
+impl LoadableYamlNode for MarkedYaml {
+    fn from_bare_yaml(yaml: Yaml) -> Self {
+        Self {
+            marker: Marker::default(),
+            data: match yaml {
+                Yaml::Real(x) => YamlData::Real(x),
+                Yaml::Integer(x) => YamlData::Integer(x),
+                Yaml::String(x) => YamlData::String(x),
+                Yaml::Boolean(x) => YamlData::Boolean(x),
+                // Array and Hash will always have their container empty.
+                Yaml::Array(_) => YamlData::Array(vec![]),
+                Yaml::Hash(_) => YamlData::Hash(LinkedHashMap::new()),
+                Yaml::Alias(x) => YamlData::Alias(x),
+                Yaml::Null => YamlData::Null,
+                Yaml::BadValue => YamlData::BadValue,
+            },
+        }
+    }
+
+    fn is_array(&self) -> bool {
+        self.data.is_array()
+    }
+
+    fn is_hash(&self) -> bool {
+        self.data.is_hash()
+    }
+
+    fn is_badvalue(&self) -> bool {
```

Comment:
I would rename this as `bad_value` (even the counterpart is `BadValue` or something along the line).

Reply:
I don't think I'm gonna rename this to `bad_value`. `Null` can sometimes be a "bad value" and that underscore would suggest that it means "is the object a [general concept of] bad value". Without underscore, `badvalue` more directly refers to the `BadValue` variant, imo.